### PR TITLE
Add possibility to switch to experimental plotting backend

### PR DESCRIPTION
**Pull request by nvaytet**
This adds the possibility to use the new plotting (still experimental) by switching to the `'experimental'` backend.
Example:
```Py
%matplotlib widget
import scipp as sc
import factory as fa

sc.plotting.select_backend('experimental')

da = fa.make_dense_data_array(ndim=2)

# Works with DataArray
da.plot()

# Works with Dataset or dict of DataArrays
sc.plot(fa.make_dense_dataset(ndim=1))

# To switch back to the old backend
sc.plotting.select_backend('stable')
```

**Reviewer:**
Changing the backend also updates the `.__doc__` property of the wrapper plot function. This feels quite hacky. Ideas/comments?

### CODE REVIEW

**Review by jl-wynen — APPROVED** [review]

File: src/scipp/plt/__init__.py [hunk 1]
```diff
@@ -7,7 +7,38 @@
...
+    Returns
+    -------
+    :
+        A figure.
+    """
+    if isinstance(obj, Variable):
+        raise TypeError("Plotting Variable is not currently supported.")
+    if isinstance(obj, DataArray):
+        return Figure(input_node(obj))
+    nodes = [input_node(v) for v in obj.values()]
```
Comment (SimonHeybrock) [inline src/scipp/plt/__init__.py:35]: Ignoring dict keys?

File: tests/plt/plot_test.py [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
 
 import scipp as sc
-from scipp.plt import Plot, Figure, widgets, input_node, node
+from scipp.plt import Plot, Figure, widgets, input_node, node, plot
```
Comment (SimonHeybrock) [inline tests/plt/plot_test.py:5]: I feel it is confusing to call this module `plt`. Can we find a clearer (temporary) name?

**Comment by nvaytet** [comment]
Lots of failed python tests.
It seems that the new plotting get activated when running the python tests, even though the old plotting tests are run after the new ones.
I'm not sure I understand in what order `pytest` imports things.

File: tests/plt/plot_test.py (hunk 2 above)
Comment (nvaytet) [reply tests/plt/plot_test.py:5]: I'm happy moving everything to an `experimental` subfolder. So put the code in `scipp.experimental.plotting` and name the function `scipp.experimental.plot`? Then we could add more things to the `experimental` submodule in the future, as we once discussed a while ago?

Comment (SimonHeybrock) [reply tests/plt/plot_test.py:5]: @jl-wynen Do you have an opinion?

Comment (jl-wynen) [reply tests/plt/plot_test.py]: Is the goal to eventually replace `scipp.plotting` and reuse the same name? If so, `scipp.experimental.plotting` sounds good.

**Comment by jl-wynen** [comment]
> I'm not sure I understand in what order pytest imports things.

If this is an issue, then we have to find a different solution. We cannot rely on a specific order of tests.

I doubt that pytest spins up a new process for every test (module). So it probably imports scipp and at some point the new plotting module once and then caches them, i.e. the default Python behaviour.

So instead of activating it by importing a module, how about a function / pair of functions to switch to new / old plotting?

**Comment by nvaytet** [comment]
Something like:
```Py
from scipp import experimental
experimental.plot.activate()
experimental.plot.deactivate()
```
?

I guess the `deactivate()` function would have to do something like
```Py
from ...plotting import plot  # <- the old plot function
DataArray.plot = plot
```
Not so pretty?

**Comment by nvaytet** [comment]
@jl-wynen @SimonHeybrock  Feel free to push changes to / merge this branch while i'm away, if you'd like to have this available.

File: tests/plotting/plot_0d_test.py [hunk 3]
```diff
@@ -8,6 +8,7 @@
 import matplotlib
 
 matplotlib.use('Agg')
+sc.plotting.select_backend('stable')
```
[the remark this answers is not in the document]
Comment (nvaytet) [reply tests/plotting/plot_0d_test.py:11]: > A safer option would be an autouse fixture that sets the backend.

Good idea.

[the remark this answers is not in the document]
Comment (nvaytet) [reply tests/plotting/plot_0d_test.py:11]: Can we also put the `matplotlib.use('Agg')` in there?

[the remark this answers is not in the document]
Comment (jl-wynen) [reply tests/plotting/plot_0d_test.py]: I don't see why not

**Review by jl-wynen — APPROVED** [review]

**Pull request merged**